src/rgw/rgw_data_sync.cc

```diff
@@ -2814,8 +2814,18 @@ RGWRemoteBucketManager::RGWRemoteBucketManager(const DoutPrefixProvider *_dpp,
                                                                     source_bucket_info.bucket,
                                                                     dest_bucket))
 {
-  int num_shards = (source_bucket_info.layout.current_index.layout.normal.num_shards <= 0 ? 
-                    1 : source_bucket_info.layout.current_index.layout.normal.num_shards);
+  rgw_bucket_index_marker_info remote_info;
+  BucketIndexShardsManager remote_markers;
+  auto r = rgw_read_remote_bilog_info(dpp, conn, source_bucket_info.bucket,
+				      remote_info, remote_markers, null_yield);
+
+  if (r < 0) {
+    ldpp_dout(dpp, 0) << __PRETTY_FUNCTION__ << ":" << __LINE__
+		      << " rgw_read_remote_bilog_info: r="
+		      << r << dendl;
+    throw boost::system::system_error(ceph::to_error_code(r));
```

Comment:
it's probably worth pulling this logic out into `RGWBucketPipeSyncStatusManager::init()` to avoid the need for exceptions. that function may construct several of these in a loop, but every iteration uses the same `source_bucket` - so you should only need to fetch this info once, and pass the same `num_shards` into each

also, the goal of this change was to avoid using `num_shards` from the local bucket info (like `source_bucket_info.layout.current_index.layout.normal.num_shards`), so please replace those below as well

the code below also does some unnecessary stuff to handle a case where `num_shards = -1`, but it should be safe to assume that `num_shards` in a layout will always be >= 1